airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcDestinationHandler.java

```diff
@@ -0,0 +1,118 @@
+import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition;
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationHandler;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId;
+import java.sql.DatabaseMetaData;
+import java.sql.JDBCType;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLType;
+import java.time.Instant;
+import java.util.LinkedHashMap;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Slf4j
+public class JdbcDestinationHandler implements DestinationHandler<TableDefinition> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcDestinationHandler.class);
+
+  protected final String databaseName;
+  protected final JdbcDatabase jdbcDatabase;
+
+  public JdbcDestinationHandler(final String databaseName,
+                                final JdbcDatabase jdbcDatabase) {
+    this.databaseName = databaseName;
+    this.jdbcDatabase = jdbcDatabase;
+  }
+
+  @Override
+  public Optional<TableDefinition> findExistingTable(final StreamId id) throws Exception {
+    return findExistingTable(jdbcDatabase, databaseName, id.finalNamespace(), id.finalName());
+  }
+
+  @Override
+  public boolean isFinalTableEmpty(final StreamId id) throws Exception {
+    final int rowCount = jdbcDatabase.queryInt(
+        """
+        SELECT row_count
+        FROM information_schema.tables
+        WHERE table_catalog = ?
+          AND table_schema = ?
+          AND table_name = ?
+        """,
+        databaseName,
+        id.finalNamespace(),
+        id.finalName());
+    return rowCount == 0;
+  }
+
+  @Override
+  public Optional<Instant> getMinTimestampForSync(final StreamId id) throws Exception {
+    return Optional.empty();
+  }
+
+  @Override
+  public void execute(final String sql) throws Exception {
+    if (sql == null || sql.isEmpty()) {
+      return;
+    }
+    final UUID queryId = UUID.randomUUID();
+    LOGGER.info("Executing sql {}: {}", queryId, sql);
+    final long startTime = System.currentTimeMillis();
+
+    try {
+      jdbcDatabase.execute(sql);
+    } catch (final SQLException e) {
+      LOGGER.error("Sql {} failed", queryId, e);
+      throw e;
+    }
+
+    LOGGER.info("Sql {} completed in {} ms", queryId, System.currentTimeMillis() - startTime);
+  }
+
+  public static Optional<TableDefinition> findExistingTable(final JdbcDatabase jdbcDatabase,
+                                                            final String databaseName,
+                                                            final String schemaName,
+                                                            final String tableName)
+      throws SQLException {
+    final DatabaseMetaData metaData = jdbcDatabase.getMetaData();
+    // TODO: normalize namespace and finalName strings to quoted-lowercase (as needed. Snowflake
```

Comment:
I feel ok ignoring this for now. Maybe eventually we can move snowflake to use the jdbc DV2 stuff, but... that's a problem for future us I think?

Reply:
I wonder if the caller rely on NamingResolver like pattern like we do in other places